### PR TITLE
Feature/refactor cubes transformations

### LINKED ISSUE #329: Refactor Rollup, DataCube, StreamingConext
New Sdk and Aggregator in order to the new policy.

### PR DESCRIPTION
**Pull request by gserranojc**
#### Description
Refactor Sparkta Policies with new Dtos.
New Cube and MultiCube implementation.
New functionality in Sdk for Dimensions, Precisions and Fields.
StreamingContextService is refactored for new sdk and aggregation engine.
New Transformations options with inputfields, outputfieds and order.
New Dimensions Types implementations.
New Dataframes schema creation.
All policies are refactored for the new implentation. 

It closes #329 #183 #255 #303 #205 #325

#### Reviewer

All

#### Test
All passed

#### Scalastyle
All test passed.

### CODE REVIEW

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 1]
```diff
@@ -173,23 +181,21 @@ object SparktaJob {
     s"$cubeName${operatorDto.measureName}"
 
   def cubes(apConfig: AggregationPoliciesDto,
-            operators: Map[String, Operator],
-            outputs: Seq[Output],
-            dimensionsMap: Map[String, Dimension]): Seq[Cube] =
+            operators: Map[String, Operator]): Seq[Cube] =
     apConfig.cubes.map(cube => {
       val name = cube.name
       val multiplexer = Try(cube.multiplexer.toBoolean).getOrElse(false)
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: This default value is so hidden, maybe this behaviour must be in the cubeDTO

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 2]
```diff
@@ -51,12 +51,10 @@ class StreamingContextService(generalConfig: Config, jars: Seq[File]) extends SL
     val inputs: Map[String, DStream[Event]] = SparktaJob.inputs(apConfig, ssc)
     val parsers: Seq[Parser] = SparktaJob.parsers(apConfig)
     val operators: Map[String, Operator] = SparktaJob.operators(apConfig)
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: Operators should be created with the cube, not before

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 3]
```diff
@@ -69,7 +67,7 @@ class StreamingContextService(generalConfig: Config, jars: Seq[File]) extends SL
       )))
 
     val bcCubeOperatorSchema: Option[Broadcast[Seq[TableSchema]]] = {
-      val rollOpSchema = PolicyFactory.cubesOperatorsSchemas(cubes, outputsSchemaConfig)
+      val rollOpSchema = SchemaFactory.cubesOperatorsSchemas(cubes, outputsSchemaConfig)
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: rollOpSchema is CubeSchema, no?

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 4]
```diff
@@ -80,7 +78,7 @@ class StreamingContextService(generalConfig: Config, jars: Seq[File]) extends SL
     SparktaJob.saveRawData(apConfig, input)
     val parsed = SparktaJob.applyParsers(input, parsers)
 
-    val dataCube = new MultiCube(dimensionsSeq, cubes, datePrecision, apConfig.checkpointing.granularity).setUp(parsed)
+    val dataCube = new MultiCube(cubes, datePrecision, apConfig.checkpointing.granularity).setUp(parsed)
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: What is multiCube? maybe is a "cubemaker" or a simple aggregation processor

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 5]
```diff
@@ -80,7 +78,7 @@ class StreamingContextService(generalConfig: Config, jars: Seq[File]) extends SL
...
     outputs.map(_._2.persist(dataCube))
     ssc
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: a map with a Unit return function could be a foreach

File: driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala [hunk 6]
```diff
@@ -91,9 +89,12 @@ object SparktaJob {
   val OperatorNamePropertyKey = "name"
 
   @tailrec
-  def applyParsers(input: DStream[Event], parsers: Seq[Parser]): DStream[Event] = parsers.headOption match {
-    case Some(parser: Parser) => applyParsers(input.map(event => parser.parse(event)), parsers.drop(1))
-    case None => input
+  def applyParsers(input: DStream[Event], parsers: Seq[Parser]): DStream[Event] = {
+    val parsersSorted = parsers.sortWith((parser1, parser2) => parser1.getOrder < parser2.getOrder)
+    parsersSorted.headOption match {
```
Comment (arinconstrio) [inline driver/src/main/scala/com/stratio/sparkta/driver/service/StreamingContextService.scala]: we are sorting the parsers inside a recursive function, has it sense?

File: plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala [hunk 7]
```diff
@@ -56,43 +56,43 @@ case class GeoHashField(props: Map[String, JSerializable])
...
-          val latLongString = value.asInstanceOf[Option[_]].get.asInstanceOf[String].split("__")
-          if (latLongString.size != 0) {
-            val latDouble = latLongString(0).toDouble
-            val longDouble = latLongString(1).toDouble
-            precision._2 -> GeoHashField.getPrecision(latDouble, longDouble, precision._2)
-          } else (precision._2 -> "")
-        })
+        val precisionKey = precision(keyName)
+        val latLongString = value.asInstanceOf[Option[_]].get.asInstanceOf[String].split("__")
+        if (latLongString.size != 0) {
```
Comment (arinconstrio) [inline plugins/field-geoHash/src/main/scala/com/stratio/sparkta/plugin/field/geohash/GeoHashField.scala]: pls refactor this method

File: driver/src/test/resources/policies/ISocket-OMongo.json [hunk 8]
```diff
@@ -52,18 +71,17 @@
           "measureName": "sum-operator",
```
Comment (sgomezg) [inline driver/src/test/resources/policies/ISocket-OMongo.json]: Rename it to ```name```, please

**Comment by gserranojc** [comment]
Operators Map now is deleted, this is the reason of the bug task not Serializable.

**Pull request merged**